File: attentive-android-sdk/src/test/java/com/attentive/androidsdk/ParameterValidationTest.java
package com.attentive.androidsdk;

import java.util.Collection;
import java.util.Collections;
import org.junit.Test;

public class ParameterValidationTest {
    @Test
    public void verifyNotNull_null_doesntThrowException() {
        ParameterValidation.verifyNotNull(null, "paramName");
    }

    @Test
    public void verifyNotNull_notNull_succeeds() {
        ParameterValidation.verifyNotNull("notNull", "paramName");
    }

    @Test
    public void verifyNotEmpty_emptyString_doesntThrowException() {
        ParameterValidation.verifyNotEmpty("", "paramName");
    }

    @Test
    public void verifyNotEmpty_nullString_doesntThrowException() {
        String nullString = null;
        ParameterValidation.verifyNotEmpty(nullString, "paramName");
    }

    @Test
    public void verifyNotEmpty_notEmptyString_succeeds() {
        ParameterValidation.verifyNotEmpty("notEmpty", "paramName");
    }

    @Test
    public void verifyNotEmpty_emptyCollection_doesntThrowException() {
        ParameterValidation.verifyNotEmpty(Collections.emptyList(), "paramName");
    }

    @Test
    public void verifyNotEmpty_nullCollection_doesntThrowException() {
        Collection<?> nullCollection = null;
        ParameterValidation.verifyNotEmpty(nullCollection, "paramName");
    }

    @Test
    public void verifyNotEmpty_notEmptyCollection_succeeds() {
        ParameterValidation.verifyNotEmpty(Collections.singletonList("notEmpty"), "paramName");
    }
}

File: attentive-android-sdk/src/test/java/com/attentive/androidsdk/ParameterValidationTest.kt
package com.attentive.androidsdk

import org.junit.Assert
import org.junit.Test

class ParameterValidationTest {
    @Test
    fun verifyNotNull_null_throwsException() {
        Assert.assertThrows(
            IllegalArgumentException::class.java,
        ) { ParameterValidation.verifyNotNull(null, "paramName") }
    }

    @Test
    fun verifyNotNull_notNull_succeeds() {
        ParameterValidation.verifyNotNull("notNull", "paramName")
    }

    @Test
    fun verifyNotEmpty_emptyString_throwsException() {
        Assert.assertThrows(
            IllegalArgumentException::class.java,
        ) { ParameterValidation.verifyNotEmpty("", "paramName") }
    }

    @Test
    fun verifyNotEmpty_notEmptyString_succeeds() {
        ParameterValidation.verifyNotEmpty("notEmpty", "paramName")
    }

    @Test
    fun verifyNotEmpty_emptyCollection_throwsException() {
        Assert.assertThrows(
            IllegalArgumentException::class.java,
        ) {
            ParameterValidation.verifyNotEmpty(
                emptyList<Any>(),
                "paramName",
            )
        }
    }

    @Test
    fun verifyNotEmpty_notEmptyCollection_succeeds() {
        ParameterValidation.verifyNotEmpty(listOf("notEmpty"), "paramName")
    }
}
